lightning/src/ln/channelmanager.rs

```diff
@@ -3591,27 +3595,22 @@ where
 										for htlc in htlcs.iter() {
 											total_value += htlc.sender_intended_value;
 											earliest_expiry = cmp::min(earliest_expiry, htlc.cltv_expiry);
-											match &htlc.onion_payload {
-												OnionPayload::Invoice { .. } => {
-													if htlc.total_msat != $payment_data.total_msat {
-														log_trace!(self.logger, "Failing HTLCs with payment_hash {} as the HTLCs had inconsistent total values (eg {} and {})",
-															log_bytes!(payment_hash.0), $payment_data.total_msat, htlc.total_msat);
-														total_value = msgs::MAX_VALUE_MSAT;
-													}
-													if total_value >= msgs::MAX_VALUE_MSAT { break; }
-												},
-												_ => unreachable!(),
+											if htlc.total_msat != $total_msat {
```

Comment:
great idea - I ended up just checking against `claimable_payment.purpose` instead of the onion payload of each htlc + was able to consolidate some other checks

also ended up being able to drop `$is_keysend` from being passed in and now get it from `claimable_htlc.onion_payload`